### ISSUE TITLE
[!] No podspec found for `RNInAppReviewIOS.xcodeproj` in `../../../node_modules/react-native-in-app-review/ios`

### ISSUE THREAD

**Issue by ionutmiftode** [MinaSamir11/react-native-in-app-review#14]
Hello,

First of all, nice job with the package! For Android works really well, my problem is on iOS.

I have a monorepo structure and when running `pod install` I receive the following error 
`[!] No podspec found for RNInAppReviewIOS.xcodeproj in ../../../node_modules/react-native-in-app-review/ios`

I tried also [manual linking](https://reactnative.dev/docs/linking-libraries-ios) but no luck I receive `/react-native/React/Base/RCTBridgeModule.h:10:9: 'React/RCTDefines.h' file not found`. 
I tried to fix this with other solutions out there like [this one](https://github.com/facebook/react-native/issues/26754#issuecomment-539550146) and [this one](https://github.com/joltup/rn-fetch-blob/issues/461#issuecomment-553355501)

Any ideas?

**Comment by MinaSamir11** [comment]
Thank you. 
Can you show your project environment?
Ex. Versions.

**Comment by ionutmiftode** [comment]
"react-native": "0.63.2",
"react-native-in-app-review": "^2.1.2",

**Comment by MinaSamir11** [comment]
React native 0.63.2, doesn't need to link manually.
Did you forget to run ```npx pod-install``` or ``` cd ios && pod install```

**Comment by ionutmiftode** [comment]
for us the automatic linking is not working as we have a monorepo structure. basically all the node modules live 2 layers above

**Comment by MinaSamir11** [comment]
I hope new release, will solve your issue !!! install version 2.1.5 and check it...

**Comment by ionutmiftode** [comment]
@MinaSamir11 still the same :( 
![Screenshot 2020-12-12 at 11 07 03](https://user-images.githubusercontent.com/6890762/101979959-45317080-3c6a-11eb-91ff-a2f36386a042.png)

**Comment by MinaSamir11** [comment]
another way,  Try to run cmd ```npx pod-install``` in the root of your project.
and,
what's your pod version ?.  run ```pod --version```

**Comment by MinaSamir11** [comment]
- If you do not have CocoaPods already installed on your machine, run ```gem install cocoapods``` to set it up the first time. (Hint: Go grab a cup of coffee!), after that run ```npx pod-install```

# another sol. :
- If you are not using Cocoapods in your project already, run cd ios && pod init at the root directory of your project. This would create a Podfile in your ios directory.
- Run react-native link react-native-in-app-review at the root directory of your project and ensure you edit your Podfile to look like the sample below (remove all the targets you are not building for, such as Tests and tvOS):

**Comment by MinaSamir11** [comment]
# Recommended 
- If you do not have CocoaPods already installed on your machine, run ```gem install cocoapods``` to set it up the first time. (Hint: Go grab a cup of coffee!), after that run ```npx pod-install```

# another sol. :
- If you are not using Cocoapods in your project already, run cd ios && pod init at the root directory of your project. This would create a Podfile in your ios directory.
- Run react-native link react-native-in-app-review at the root directory of your project and ensure you edit your Podfile to look like the sample below (remove all the targets you are not building for, such as Tests and tvOS):

```
target '_YOUR_PROJECT_TARGET_' do

  # Pods for _YOUR_PROJECT_TARGET_
....
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  # add this
  pod 'react-native-in-app-review', :path => '../node_modules/react-native-in-app-review'

```

- By now, you should be all set to install the packages from your Podfile. Run ```pod install``` from your ios directory.

- Close Xcode, and then open (double-click) your project's .xcworkspace file to launch Xcode. From this time onwards, you must use the .xcworkspace file to open the project. Or just use the react-native run-ios command as usual to run your app in the simulator.

**Comment by ionutmiftode** [comment]
![Screenshot 2020-12-12 at 14 05 11](https://user-images.githubusercontent.com/6890762/101983332-24c1e000-3c83-11eb-9d40-40152cfc070e.png)
I'll the other solution also and I'll let you know

**Comment by ionutmiftode** [comment]
Still no luck :( 
![Screenshot 2020-12-12 at 15 37 29](https://user-images.githubusercontent.com/6890762/101985677-3ced2b80-3c92-11eb-9dab-2fb0c6342816.png)
![Screenshot 2020-12-12 at 15 51 56](https://user-images.githubusercontent.com/6890762/101985688-49718400-3c92-11eb-8b5a-399dd73aaa20.png)
![Screenshot 2020-12-12 at 15 52 45](https://user-images.githubusercontent.com/6890762/101985693-51c9bf00-3c92-11eb-9123-81540883c08e.png)

**Comment by bodrius** [comment]
the same problem

**Comment by MinaSamir11** [comment]
@ionutmiftode  can you share your structure  of your monorepo please?

**Comment by MinaSamir11** [comment]
> the same problem

you are using version 2.1.4 OR 2.1.5 ?

**Comment by bodrius** [comment]
V 2.1.5

**Comment by ionutmiftode** [comment]
Sure! attaching the structure

![Screenshot 2020-12-14 at 18 01 26](https://user-images.githubusercontent.com/6890762/102104037-647cf900-3e36-11eb-89be-06f7c2ca6f4a.png)

**Comment by bodrius** [comment]
![Снимок экрана 2020-12-14 в 18 05 07](https://user-images.githubusercontent.com/57833393/102104717-30ee9e80-3e37-11eb-899d-7088bd3f5893.png)

**Comment by MinaSamir11** [comment]
> Sure! attaching the structure
> 
> ![Screenshot 2020-12-14 at 18 01 26](https://user-images.githubusercontent.com/6890762/102104037-647cf900-3e36-11eb-89be-06f7c2ca6f4a.png)

I noticed that their is  multiple node_modules in the project, i was just wondering where is react-native-in-app-review installed and if its possible to provide me with copy of your mobile app package.json

on package.json founded in the root and on package.json in packages/mobile for just debugging.

**Comment by MinaSamir11** [comment]
@bodrius  can you install version 2.1.4 please and check if problem still happen or not ?

**Comment by ionutmiftode** [comment]
Root package.json /Users/mifi/work/wewash/package.json
mobile package.json /Users/mifi/work/wewash/packages/mobile/package.json
We install all dependencies as root as we have multiple packages that use them

**Comment by ionutmiftode** [comment]
Hello @MinaSamir11  any updates on this?

**Comment by MinaSamir11** [comment]
> Hello @MinaSamir11  any updates on this?

I have technical issue this days forgive me for late, i'm working in your issue and i'm trying to test my solution before deliver it to you.

**Comment by ionutmiftode** [comment]
I'm for your technical issue but I'm happy that soon we will have a fix :D

**Comment by MinaSamir11** [comment]
Can we try out some solution and give me feedback.
and can we connect directly on linked to chat to reach to the sol.

https://www.linkedin.com/in/mina-samir-925587145

**Comment by MinaSamir11** [comment]
@bodrius issue resolved now in version @2.1.8 according to @ionutmiftode .
 
try with new version before reinstall clear cache with this cmd 

```
npm start -- --reset-cache 
````

_I'm closing these issue because it resolved in version @2.1.8 ._
